### PR TITLE
chore: complete products with filters

### PR DESCRIPTION
**Pull request by Mohitkumar33**
Completed products page,
products with maximum discount are loading on home page
filter complete
used useEffect to get the data globally using use context

### CODE REVIEW

File: src/pages/Products.js [hunk 1]
```diff
@@ -1,11 +1,368 @@
...
 
-const Products = ()=>{
-    return(
-        <>
-        <p>This is Products</p>
-        </>
+const Products = () => {
+  const { productsState } = useProducts();
+  const { filterState, filterDispatch } = useFilters();
+  console.log(productsState.productsState);
```
Comment (Richa-rani02) [inline src/pages/Products.js:20]: Remove console

File: src/utilitites/filterByGender.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  ) {
+    return data;
+  } else if (menInState === true) {
+    return data.filter((i) => i.gender === "men");
+  } else if (womenInState === true) {
+    return data.filter((i) => i.gender === "women");
+  }
+};
+
+// const filterByMen = (menInState, data) => {
```
Comment (Richa-rani02) [inline src/utilitites/filterByGender.js:17]: Remove commeted code

File: src/utilitites/filterBySeason.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+// const filterByWinter = (winterInState, data) => {
+//   if (winterInState === null) {
+//     return data;
+//   } else if (winterInState === true) {
+//     return data.filter((i) => i.season === "winter");
+//   }
+//   return data;
+// };
+
```
Comment (Richa-rani02) [inline src/utilitites/filterBySeason.js:34]: Here as well

**Review by Richa-rani02 — CHANGES_REQUESTED** [review]

**Review by mnath30 — APPROVED** [review]
Code is clean
The variable naming is good
Just remove the commented code if not required

**Review by tanishk15sharma — APPROVED** [review]

**Comment by Mohitkumar33** [comment]
Thanks all for PR review

**Pull request merged**